apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -259,131 +233,54 @@ class TeacherFeedback extends Component {
-                  {showFeedbackInputAreas && (
-                    <input
-                      type={'checkbox'}
-                      id={'meets-input'}
-                      name={'rubric'}
-                      value={'meets'}
-                      checked={this.state.performance === 'meets'}
-                      onChange={this.onRubricChange}
-                      disabled={this.props.disabledMode}
-                    />
-                  )}
-                  <details>
-                    <summary style={styles.rubricHeader}>Meets</summary>
-                    <p>{this.props.rubric.meets}</p>
-                  </details>
-                </div>
-                <div
-                  style={
-                    this.state.performance === 'approaches'
-                      ? styles.performanceLevelHeaderSelected
-                      : styles.performanceLevelHeader
-                  }
-                >
-                  {showFeedbackInputAreas && (
-                    <input
-                      type={'checkbox'}
-                      id={'approaches-input'}
-                      name={'rubric'}
-                      value={'approaches'}
-                      checked={this.state.performance === 'approaches'}
-                      onChange={this.onRubricChange}
-                      disabled={this.props.disabledMode}
-                    />
-                  )}
-                  <details>
-                    <summary style={styles.rubricHeader}>Approaches</summary>
-                    <p>{this.props.rubric.approaches}</p>
-                  </details>
-                </div>
-                <div
-                  style={
-                    this.state.performance === 'noEvidence'
-                      ? styles.performanceLevelHeaderSelected
-                      : styles.performanceLevelHeader
-                  }
-                >
-                  {showFeedbackInputAreas && (
-                    <input
-                      type={'checkbox'}
-                      id={'noEvidence-input'}
-                      name={'rubric'}
-                      value={'noEvidence'}
-                      checked={this.state.performance === 'noEvidence'}
-                      onChange={this.onRubricChange}
-                      disabled={this.props.disabledMode}
-                    />
-                  )}
-                  <details>
-                    <summary style={styles.rubricHeader}>No Evidence</summary>
-                    <p>{this.props.rubric.noEvidence}</p>
-                  </details>
-                </div>
+                <RubricField
+                  showFeedbackInputAreas={showFeedbackInputAreas}
+                  rubricLevel={'exceeds'}
+                  rubricValue={this.props.rubric.exceeds}
+                  disabledMode={this.props.disabledMode}
+                  onChange={this.onRubricChange}
+                  currentlyChecked={this.state.performance === 'exceeds'}
```

Comment:
You're very close to being able to define these all in a loop:

```js
const rubricFields = ['exceeds', 'meets', 'approaches', 'noEvidence'];
```
and then
```js
<form>
  {
    rubricLevels.map(level => (
      <RubricField
        showFeedbackInputAreas={showFeedbackInputAreas}
        rubricLevel={level}
        rubricValue={this.props.rubric[level]}
        disabledMode={this.props.disabledMode}
        onChange={this.onRubricChange}
        currentlyChecked={this.state.performance === level}
      />
    ))
  }
</form>
```

Reply:
I'd recommend the above approach. but if you'd like to see a close alternative, one variation would be to  change the following two props to be more generic:
```js
        rubricValues={this.props.rubric}
        performance={this.state.performance}
```
then inside RubricField, you can compute the values you need like:
```js
const {rubricLevel, rubricValues, performance} = this.props;
const rubricValue = rubricValues[rubricLevel];
const currentlyChecked = (performance == rubricLevel);
```